src/main/java/ca/wolfram/beta/symath/operations/AddNode.java

```diff
@@ -35,19 +35,49 @@ public NodeType getType() {
         return NodeType.ADD;
     }
 
+    /**
+     * Extracts all constant integers and appends it as one {@link ca.wolfram.beta.symath.base.ConstantNode} if the sum != 0
+     * Flattens children if they are also of type ADD
+     *
+     * @return true is isConstant, false otherwise
+     */
     @Override
     public boolean simplify() {
-        int constant = 0;
-        final Iterator<MathNode> each = getChildren().iterator();
+        int constant = simplify(0, 0);
+        if (constant != 0)
+            children.add(BaseNode.create(constant));
```

Comment:
i feel we should be consistent between using getChildren() and just calling children. getChildren() is better i think